### PR TITLE
DAOS-7808 test: Clean up EC, checksum, datamover tests

### PR DESCRIPTION
**Pull request by sylviachanoiyee**
Standardize class naming convention in components using 4 character prefix.
Fix tagging.

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]
LGTM. Though, with the class names being changed, I wonder if the test case names should be reflective, and the tags as well? For example:
```
class DmObjSmallTest -> class DmvrObjSmallTest
    ...
    def test_dm_obj_small_dcp -> def test_dmvr_obj_small_dcp
        ...
        :avocado: tags=datamover,dmvr,dcp
        :avocado: tags=dmvr_obj_small,dmvr_obj_small_dcp
```

File: src/tests/ftest/erasurecode/ec_ior_smoke.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 '''
 from ior_test_base import IorTestBase
 
-class ErasureCodeIor(IorTestBase):
+class EcodIor(IorTestBase):
```
Comment (ravalsam) [inline src/tests/ftest/erasurecode/ec_ior_smoke.py:9]: It should be EcodeIor? missing "e" in code or it's intended?

Comment (daltonbohning) [reply src/tests/ftest/erasurecode/ec_ior_smoke.py:9]: I think the idea here is to use exactly 4 letters. E.g. "Ecod" and "Dmvr"

Comment (ravalsam) [reply src/tests/ftest/erasurecode/ec_ior_smoke.py:9]: Ah got it.

**Review by ravalsam — APPROVED** [review]

**Review by rpadma2 — DISMISSED** [review]

**Comment by sylviachanoiyee** [comment]
The pool destroy rebuild test failure in build #1 is not related to changes in this PR.
This PR renamed files and class under erasurecode, checksum and datamover. No changes under ftests/rebuild.
Test was rerun and passing in build#2
Issue filed as 
https://jira.hpdd.intel.com/browse/DAOS-7815

**Review by ravalsam — APPROVED** [review]

**Comment by sylviachanoiyee** [comment]
> LGTM. Though, with the class names being changed, I wonder if the test case names should be reflective, and the tags as well? For example:
> 
> ```
> class DmObjSmallTest -> class DmvrObjSmallTest
>     ...
>     def test_dm_obj_small_dcp -> def test_dmvr_obj_small_dcp
>         ...
>         :avocado: tags=datamover,dmvr,dcp
>         :avocado: tags=dmvr_obj_small,dmvr_obj_small_dcp
> ```

That's a good suggestion for maintaining consistency. Maybe we can do that in a different pr when we scrub tags more thoroughly

**Pull request merged**